transform-documents.py

```
@@ -412,35 +416,28 @@ async def main():
-            await process_file(txt, parser, handler)
+        logger.info(f"No .txt files found under {input_dir}")
+        sys.exit(1)
 
-    # 7) read concurrency setting
+    # 7. read concurrency setting
     try:
         max_conc = int(os.getenv("MAX_CONCURRENT", "0"))
     except ValueError:
-        print("Error: MAX_CONCURRENT must be an integer")
-        sys.exit(1)
+        raise ValueError("MAX_CONCURRENT must be an integer")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)